sdcm/nemesis.py

```diff
@@ -1841,12 +1842,18 @@ def repair_nodetool_rebuild(self):
 
     def disrupt_nodetool_cleanup(self):
         # This fix important when just user profile is run in the test and "keyspace1" doesn't exist.
-        test_keyspaces = self.cluster.get_test_keyspaces()
-        for node in self.cluster.nodes:
+        # test_keyspaces = self.cluster.get_test_keyspaces()
+
+        # Inner disrupt function for ParallelObject
+        def _dis_nodetool_cleanup(node):
             InfoEvent('NodetoolCleanupMonkey %s' % node).publish()
             with adaptive_timeout(Operations.CLEANUP, node, timeout=HOUR_IN_SEC * 48):
-                for keyspace in test_keyspaces:
-                    node.run_nodetool(sub_cmd="cleanup", args=keyspace)
+                # for keyspace in test_keyspaces:
+                node.run_nodetool(sub_cmd="cleanup")
+
+        threads = ParallelObject(self.cluster.nodes, num_workers=min(32, len(self.cluster.nodes)), timeout=43200)
+        thread_results = threads.run(_dis_nodetool_cleanup, unpack_objects=True)
+        wait_for_futures(thread_results)
```

Comment:
ParallelObject is waiting for the futures, I'm not sure why it's needed to wait for them again ?

Reply:
It does not wait as it appears from here:
file:///home/mark/logs/NodeToolCleanupMonkey-ParallelObject%235/parallel-timelines-report-4ce134c7/longevity-5gb-1h-NodeToolCleanupMon-4ce134c7-parallel-timelines-report.html

This `wait()` also doesn't work as ParallelObjectResult has no required attributes. It does not inherit from `Future` class in concurrent.futures lib.

Reply:
as I said ParallelObject should be do the waiting on it's own

how come you get to situation it returns without all commands finished its not clear to me, it mean you are not passing a timeout to the nodetool command, also disabling the retries is recommended.

Reply:
just remove `wait_for_futures(thread_results)` and I think it would work o.k., now that you have 12h timeout

Reply:
ok